src/transformers/configuration_utils.py

```diff
@@ -55,7 +53,6 @@ class PretrainedConfig(object):
 
     def __init__(self, **kwargs):
         # Attributes with defaults
-        self.output_attentions = kwargs.pop("output_attentions", False)
```

Comment:
@thomwolf @LysandreJik @sshleifer - `output_attentions` is completely removed as an argument to the config. This means that `output_attentions` always defaults to `False` in the forward functions and cannot be influenced anymore by the `config`. 

It is arguable whether we should instead allow to initialize `config.output_attentions` with a boolean and let all `output_attentions` in all `forward()` functions default to `None` and instead take the value from `config.output_attentions`, but there would be some problems here as well:

1) It adds a lot of code
2) It's probably confusing to the user to read in the docs that the boolean `output_attentions` argument is init to `None` for all forward functions, but behaves as init to `False`. 
3) At which abstraction level do we still allow the `output_attentions` to be initialized from the config? `BertModel.forward()` should probably be initialized with `config.output_attentions`, but `BertAttentionLayer()` as well? How long do we have to pass down the config?
4) This could also introduce the following confusion: Contrary to the `generate()` method which directly relies on the `self.config` config, some models, e.g. T5 make a copy of the original config and then pass the config further down. This could lead to confusion when people initially instantiate a T5 model with `config.output_attentions=True` and try to change it afterward with `model.config.output_attentions=False`, which would not work, since the model classes are already initiated with `config.output_attentions=True`. 

All in all, I would be in favor of removing `config.output_attentions` and initialize `output_attentions` only via the argument `output_attentions=False` in all forward functions. I think the same should apply for `use_cache` and `output_hidden_states` (`use_cache` is already completely independent for all `forward()` methods and only the `generate()` method relies on the config, which I think we could change as well. 
IMO `use_cache`, `output_attentions` and `output_hidden_states` all are not really parameters of the model or its decoding strategy, but switches for all (or some for `use_cache`)models that can be turned on and off, but essentially don't influence the model's output logits. In contrast the other config attributes ("max_length", "do_sample", ...) do influence the output of the model and therefore should stay in the config. Therefore I would be ok with removing `use_cache`, `output_attentions` and `output_hidden_states` completely from the `config`. 

But that's up for debate!

What is your opinion on that?